From ad0c86adea108698b41503a994a0c70d9ae99bc6 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Wed, 23 Jun 2021 21:18:47 -0500
Subject: [PATCH] [WIP] ASoC: sun20i-codec: What is this ramp thing?

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 sound/soc/sunxi/sun20i-codec.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sound/soc/sunxi/sun20i-codec.c b/sound/soc/sunxi/sun20i-codec.c
index 34009d60f54b2f..110c9a40e07401 100644
--- a/sound/soc/sunxi/sun20i-codec.c
+++ b/sound/soc/sunxi/sun20i-codec.c
@@ -709,6 +709,10 @@ static int sun20i_codec_component_probe(struct snd_soc_component *component)
 				      0x3 << SUN20I_CODEC_ADC_DIG_CTRL_ADC_VOL_EN,
 				      0x3 << SUN20I_CODEC_ADC_DIG_CTRL_ADC_VOL_EN);
 
+	/* Maaagic... */
+	snd_soc_component_update_bits(component, SUN20I_CODEC_RAMP,
+				      BIT(1) | BIT(0), BIT(0));
+
 	return 0;
 
 err_assert_reset:
